.github/workflows/check_correctness.yml

```diff
@@ -0,0 +1,72 @@
+# This is a basic workflow to help you get started with Actions
+
+name: check_correctness
+
+# Controls when the action will run. Triggers the workflow on push or pull request
+# events but only for the master branch
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: '**'
+  pull_request:
+    branches: '**'
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+
+      - name: Opam modules cache
+        uses: actions/cache@v1
+        env:
+          cache-name: cache-opam-modules
+        with:
+          # OCaml cache files are stored in `~/.opam` on Linux/macOS
+          path: ~/.opam
```

Comment:
If you want to stick with `make deps`, local opam switches seem to install in `_opam`  in the project directory rather than `~/.opam`.

From reading the documentation something like this should work : 
```yaml
path: $GITHUB_WORKSPACE/_opam
```